static/js/presence.js

```diff
@@ -38,7 +38,11 @@ exports.get_status = function (user_id) {
 };
 
 exports.get_user_ids = function () {
-    const user_ids = Object.keys(exports.presence_info);
+    let user_ids = Object.keys(exports.presence_info);
+    user_ids = _.map(user_ids, function (s) {
+        return parseInt(s, 10);
+    });
+
```

Comment:
Prefer `const user_ids = Object.keys(exports.presence_info).map(s => parseInt(s, 10))`. The type-shifting mutation is unnecessary, and TypeScript won’t like it when it becomes TypeScript.

Perhaps `presence_info` should be `IntDict` to avoid this conversion entirely?

Reply:
Oops, I missed this one.  I'm not sure presence is the best candidate for `IntDict`, since it's a short-lived data structure coming from the server as raw JSON every minute or so.  Gonna fix the `let`.